llvm-bazel/terminfo.bzl

```diff
@@ -0,0 +1,217 @@
+                linkopts = linkopts_candidate
+                break
+
+        # If we never found a viable linkopts candidate, either error or disable
+        # terminfo for LLVM.
+        if not linkopts:
+            if required:
+                fail("Failed to detect which linkopt would successfully provide the " +
+                    "necessary terminfo functionality")
+            else:
+                _llvm_terminfo_disable_impl(repository_ctx)
+                return
+
+    repository_ctx.template(
+        "BUILD",
+        repository_ctx.attr._system_build_template,
+        substitutions = {
+            "{TERMINFO_LINKOPTS}": str(linkopts),
+        },
+        executable = False,
+    )
+
+_terminfo_system_attrs = {
+    "_disable_build_template": attr.label(
+        default = Label("//:terminfo_system.BUILD"),
+        allow_single_file = True,
+    ),
+    "_system_build_template": attr.label(
+        default = Label("//:terminfo_system.BUILD"),
+        allow_single_file = True,
+    ),
+    "_terminfo_test_source": attr.label(
+        default = Label("//:terminfo_test.c"),
+        allow_single_file = True,
+    ),
+    "candidate_system_linkopts": attr.string_list(
+        default = ["-lterminfo", "-ltinfo", "-lcurses", "-lncurses",
+                    "-lncursesw"],
+        doc = "Candidate linkopts to test and see if they can link " +
+              "successfully.",
+    ),
+    "system_required": attr.bool(
+        default = False,
+        doc = "Require that one of the candidates is detected successfully on POSIX platforms where it is needed.",
+    ),
+    "system_linkopts": attr.string_list(
+        default = [],
+        doc = "If non-empty, a specific array of linkopts to use to " +
+              "successfully link against the terminfo library. No " +
+              "detection is performed if this option is provided, it " +
+              "directly forces the use of these link options. No test is " +
+              "run to determine if they are valid or work correctly either.",
+    ),
+}
+
+llvm_terminfo_system = repository_rule(
+    implementation = _llvm_terminfo_system_impl,
+    local = True,
+    attrs = _terminfo_system_attrs,
+)
+
+def _llvm_terminfo_disable_impl(repository_ctx):
+    repository_ctx.template(
+        "BUILD",
+        repository_ctx.attr._disable_build_template,
+        executable = False,
+    )
+
+_terminfo_disable_attrs = {
+    "_disable_build_template": attr.label(
+        default = Label("//:terminfo_disable.BUILD"),
+        allow_single_file = True,
+    ),
+}
+
+llvm_terminfo_disable = repository_rule(
+    implementation = _llvm_terminfo_disable_impl,
+    local = True,
+    attrs = _terminfo_disable_attrs,
+)
+
+def _llvm_terminfo_from_env_impl(repository_ctx):
+    terminfo_strategy = repository_ctx.os.environ.get("BAZEL_LLVM_TERMINFO_STRATEGY")
+    if terminfo_strategy == "external":
+        pass #_llvm_terminfo_external_impl(repository_ctx)
```

Comment:
Error?

Reply:
Sorry, removing this too didn't get saved!